service/src/main/kotlin/fi/espoo/evaka/document/childdocument/ChildDocumentController.kt

```diff
@@ -100,11 +102,18 @@ class ChildDocumentController(
                             clock,
                             documents.map { it.id }
                         )
-                    documents.mapNotNull { document ->
-                        permittedActions[document.id]
-                            ?.takeIf { it.contains(Action.ChildDocument.READ) }
-                            ?.let { ChildDocumentSummaryWithPermittedActions(document, it) }
-                    }
+                    documents
+                        .mapNotNull { document ->
+                            permittedActions[document.id]
+                                ?.takeIf { it.contains(Action.ChildDocument.READ) }
+                                ?.let { ChildDocumentSummaryWithPermittedActions(document, it) }
+                        }
+                        .filter {
+                            VASU_MIGRATION_COMPLETED ||
```

Comment:
This filter implementation could be extracted somewhere and the same implementation could be used in both `getDocument()` and `getDocuments()`. And I think the variable `VASU_MIGRATION_COMPLETED` should be removed.

Reply:
The idea I was going for here was that once the migration is considered ready, the constant value could be flipped to make the results visible to non-admins, and soon after if everything went fine all these filters could be permanently removed. But yeah maybe that's a bit unnecessary if it's just a global constant and not an env variable which would allow different cities and envs to have the go-live at different points in time.

What about the extraction then. I guess the DocumentType enum could have a boolean property telling if it's migrated. That would replace the `it.data.type !in listOf(DocumentType.MIGRATED_VASU, DocumentType.MIGRATED_LEOPS)` condition which is the only fully shared part, assuming that the `VASU_MIGRATION_COMPLETED` is removed.

Reply:
Implemented that change